src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -738,16 +739,32 @@ private void WindowClosing(object sender, CancelEventArgs e)
                 //Shutdown wasn't cancelled
                 SizeChanged -= DynamoView_SizeChanged;
                 LocationChanged -= DynamoView_LocationChanged;
+                return true;
             }
             else
             {
                 //Shutdown was cancelled
-                e.Cancel = true;
+                return false;
             }
         }
 
+        private void WindowClosing(object sender, CancelEventArgs e)
+        {
+            if (!PerformShutdownSequenceOnViewModel() && !DynamoModel.IsTestMode)
+                e.Cancel = true;
+            else
+                isPSSCalledOnViewModelNoCancel = true;
+        }
```

Comment:
This is just style, but I'd prefer if you used curly brackets around these code blocks.  This makes it harder to make silly refactoring mistakes.